assets/javascripts/modules/checkout/validations.js

```diff
@@ -28,21 +38,49 @@ define(['$', 'modules/checkout/form-elements', 'modules/checkout/regex'], functi
     }
 
     var validatePersonalDetails = function () {
-        var emptyFields = mandatoryFieldsPersonalDetails.filter(function (field) {
-            var isEmpty = field.input.val() === '';
-            toggleError(field.container, isEmpty);
-            return isEmpty;
+
+        var staticValidations = new Promise(function (onSuccess, onFailure){
+            var emptyFields = mandatoryFieldsPersonalDetails.filter(function (field) {
+                var isEmpty = field.input.val() === '';
+                toggleError(field.container, isEmpty);
+                return isEmpty;
+            });
+            var noEmptyFields = emptyFields.length === 0;
+
+            var validEmail = regex.isValidEmail(form.$EMAIL.val());
+            renderEmailError(!validEmail);
+            toggleError(form.$EMAIL_CONTAINER, !validEmail);
+
+            var emailCorrectTwice = form.$EMAIL.val() === form.$CONFIRM_EMAIL.val();
+            toggleError(form.$CONFIRM_EMAIL_CONTAINER, validEmail && !emailCorrectTwice);
+
+            if(noEmptyFields && emailCorrectTwice && validEmail){
+                onSuccess();
+            }
+            else {
+                onFailure(new Error('Personal details incorrect.'));
+            }
         });
-        var noEmptyFields = emptyFields.length === 0;
 
-        var validEmail = regex.isValidEmail(form.$EMAIL.val());
-        renderEmailError(!validEmail);
-        toggleError(form.$EMAIL_CONTAINER, !validEmail);
+        var validationResult = staticValidations.then(function(){
+
+            var warnIfEmailTaken = emailCheck.warnIfEmailTaken();
+
+            warnIfEmailTaken.catch(function (error) {
```

Comment:
I'm not sure I'd class the email being taken as an error, as that's the point of the function. Makes sense the network failure (or just failure in general) being an error, but would I'd expect the email being taken to be part of the `resolve`. Either a boolean or an object with validation information.